xla/stream_executor/stream_executor_common.h

```diff
@@ -51,6 +51,8 @@ class StreamExecutorCommon : public StreamExecutor {
 
   const Platform* GetPlatform() const override { return platform_; }
   const DeviceDescription& GetDeviceDescription() const override;
+  //To be used in tests only
+  void  SetThreadsPerWarp(const int64_t value) override;
```

Comment:
Please don't add this method in the actual interface.  What is the actual bug that is happening here?  Is the number of threads_per_warp being incorrectly returned in the DeviceDescription for the underlying hardware?  The original PR was a change to use the correct threads_per_warp for the hardware being tested.

Reply:
But most of the tests are written keeping 32 threads per warp. These tests need to be rewritten and again we may have to distinguish between AMD (64 threads/warp) and NVIDIA (I believe 32 threads/warp) hardware in the test, as they have different values. 

This is a make-shift patch until we fix the tests in a hardware agnostic way.

Reply:
majnemer has authored https://github.com/openxla/xla/pull/20170 which fixes this.

Reply:
https://github.com/openxla/xla/pull/20170 sets threads_per_warp on newer GPUs to 32 but our CI systems run on gfx910 which still get warp size of 64 and hence tests fails.
I am also not sure if we have to hardcode the values instead of relying on hipDevicGetAttributes api which would return correct warp size depending on if wavefrontsize64 is enabled or not.

Reply:
I also see a TODO from https://github.com/openxla/xla/pull/10649 saying that threads_per_warp must be 32 in the triton pipeline or things don't work.  Maybe all this should just be dropped, and rocm_executor should just set threads_per_warp to be 32.

Reply:
I believe that most of the tests are written with 32 threads/warp tiling in mind w.r.t memory allocation and the tensor layout to match that value. We may want to work on extending the tests to support 64 threads/warp as well. But hardcoding in rcom_executor to 32 may not be the right solution in the long run

Reply:
I don't think these tests have any threads_per_warp baked into them.  Many of these tests (convolution_test.cc for certain) are backend-agnostic, and run on accelerators without any notion of threads_per_warp at all.

Reply:
Since the test break is now fixed, I think this PR should be closed in favor of some future CL which fixes all the hardcoding of 32 threads_per_warp for AMD code.